jobsdb/jobsdb.go

```diff
@@ -387,22 +391,23 @@ var dbErrorMap = map[string]string{
 	"Invalid Escape Character": "22019",
 }
 
-// return backup settings depending on jobdb type
-// the gateway, the router and the processor
-// BackupEnabled = true => all the jobsdb are eligible for backup
+// registers the backup settings depending on jobdb type the gateway, the router and the processor
+// masterBackupEnabled = true => all the jobsdb are eligible for backup
 // instanceBackupEnabled = true => the individual jobsdb too is eligible for backup
 // instanceBackupFailedAndAborted = true => the individual jobdb backsup failed and aborted jobs only
 // pathPrefix = by default is the jobsdb table prefix, is the path appended before instanceID in s3 folder structure
-func (jd *HandleT) getBackUpSettings() *BackupSettingsT {
-	config.RegisterBoolConfigVariable(true, &masterBackupEnabled, false, "JobsDB.backup.enabled")
-	config.RegisterBoolConfigVariable(false, &instanceBackupEnabled, false, fmt.Sprintf("JobsDB.backup.%v.enabled", jd.tablePrefix))
-	config.RegisterBoolConfigVariable(false, &instanceBackupFailedAndAborted, false, fmt.Sprintf("JobsDB.backup.%v.failedOnly", jd.tablePrefix))
+func (jd *HandleT) registerBackUpSettings() {
+	jd.BackupSettings = &BackupSettingsT{
+		masterBackupEnabled: true,
+		instanceBackupEnabled: true,
+		FailedOnly: false,
+		PathPrefix: "",
+	}
+	config.RegisterBoolConfigVariable(true, &jd.BackupSettings.masterBackupEnabled, true, "JobsDB.backup.enabled")
```

Comment:
assuming `masterBackupEnabled` 
```suggestion
	config.RegisterBoolConfigVariable(true, &masterBackupEnabled, true, "JobsDB.backup.enabled")
```

and use `sync.Once` or move it to the Init.

Reply:
I don't understand why? Now there is no `masterBackupEnabled`. This has been done to avoid using the global variable. Though this is always going to be same for all type of jobsDBs

Reply:
is it hard to turn this
```
hotReloadableConfig    map[string]*ConfigVar
```

into that?

```
hotReloadableConfig    map[string][]*ConfigVar
```

Reply:
It should not be hard @atzoum  but we would need to test it rigorously since it might impact our ability to tweak the config's